splunk/utils.go

```diff
@@ -54,9 +54,24 @@ func (c *splunkClient) BuildSplunkURL(path string, params interface{}) url.URL {
 		q.Set("count", "0")
 	}
 
+	// By default use https
+	urlScheme := "https"
+	host := c.URL
+
+	parsedAPIURL, parseErr := url.Parse(c.URL)
```

Comment:
Are there any cases where a user could have a URL in their config today that would somehow work when connecting to Splunk's endpoints, but error on the `Parse` function? Can you please include what URLs you tested and the query results as part of this change?

Reply:
@cbruno10 I tested some of the possible cases, here is the list of the URLs:
```
"localhost:8089"
"https://localhost:8089"
"https://prd-p-ydyk6.splunkcloud.com"
"prd-p-ydyk6.splunkcloud.com"
"prd-p-ydyk6.splunkcloud.com:8089"
"http://google.com"
"http://w.com/cn"
"http://192.158.0.1:90"
"http://w"
"fsw"
"http://192.158.1/1"
```